docs/reactivity/reactive-components.md

```diff
@@ -0,0 +1,99 @@
+
+Oryx components are build in TypeScript, which is why we need to ensure type safety. The original type of the assigned observable needs to be adjusted. It's not possible to resolve correct TypeScript type from the observable by just using decorator, which is why the `valueType` function is offered to project the observed type.
+
+```ts
+export class ProductPriceComponent {
+  @asyncState()
+  protected prices = valueType(
+    this.productController
+      .getProduct()
+      .pipe(switchMap((product) => this.formatPrices(product?.price)))
+  );
+
+  protected override render(): TemplateResult | void {
+    return html`${this.prices.defaultPrice}`;
+  }
+}
+```
+
+## Operate on multiple streams
+
+Components often use multiple data streams. A good example is the product price which renders a product price in a certain currency and a _local_ price format. The currency and locale are part of the application context and might change during the application lifecycle. The product price will change from product to product. In order to combine the various streams, RxJS operators comes into play. RxJS operators can combine multiple observable and operate on the combined results.
+
+This is illustrated in the example below, where the `ProductPriceComponent` observes the product data from the `ProductService` and _combines_ this with the formatted price given by the `PriceService`.
+
+```ts
+export class ProductPriceComponent {
+  protected productService = resolve(ProductService);
+  protected priceService = resolve(PriceService);
+
+  protected prices = this.productService
+    .getProduct()
+    .pipe(switchMap((product) => this.formatPrices(product?.price)));
+
+  protected formatPrices(
+    price?: ProductPrices
+  ): Observable<{ originalPrice: string | null; salesPrice: string | null }> {
+    const salesPrice = this.priceService.format(price?.defaultPrice);
+    const originalPrice = this.priceService.format(price?.originalPrice);
+    return combineLatest({ salesPrice, originalPrice });
+  }
+}
+```
+
+**Note:** the example is intentionally simplified to keep the focus on the RxJs part.
+
+In this example code you can see that the product data is observed from the ProductService, but switches to the price formatting logic. This means that whenever the product changes (i.e. on route change), new product data is emitted and formatted. The `PriceService` is used to format both the sales and original price. The `PriceService.format()` uses the current currency and locale for the formatting, which is why it also exposes an observable. Since there are two prices involved, the two streams are _combined_ in an object.
+
+With this setup, the product price component will receive updated prices whenever the product, currency or locale in the application state will change.Subscribing to this code could simple, however more work is needed to ensure that the component UI (DOM) will be updated simultaneously. This is described in the next section.
```

Comment:
1. Please doublecheck the second sentence, smth is amiss.
2. Where is the next section?

Reply:
has already been covered before in the text, have dropped the last section.